# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/socket.h.patch
# Copyright (C) 2011 - 2014 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

fix udev:

libudev-monitor.c:157:52: error: 'SO_RCVBUFFORCE' undeclared (first use in this function)

libudev-monitor.c:142:51: error: 'SOCK_CLOEXEC' undeclared (first use in this function)


  - Rene Rebe <rene@exactcode.de>

--- dietlibc/include/sys/socket.h.vanilla	2011-01-05 14:48:32.000000000 +0100
+++ dietlibc/include/sys/socket.h	2011-01-05 14:49:11.000000000 +0100
@@ -145,6 +145,8 @@
 
 #endif
 
+#define SO_RCVBUFFORCE  33
+
 #if !defined(__hppa__) && !defined(__sparc__)
 /* Security levels - as per NRL IPv6 - don't actually do anything */
 #define SO_SECURITY_AUTHENTICATION		22
@@ -377,6 +378,9 @@
 #define SOCK_PACKET	10		/* linux specific way of	*/
 #endif
 
+#define SOCK_NONBLOCK  04000           /* Atomically mark descriptor(s) */
+#define SOCK_CLOEXEC	02000000	/* Atomically set close-on-exec flag for the new descriptor(s). */
+
 int socket(int domain, int type, int protocol) __THROW;
 int accept(int s, struct sockaddr *addr, socklen_t *addrlen) __THROW;
 int connect(int sockfd, const struct sockaddr *serv_addr, socklen_t addrlen) __THROW;
